src/components/FileEntryTree.tsx

```diff
@@ -1,72 +1,79 @@
-import { useState } from 'react';
+import { useAtom, useAtomValue } from 'jotai';
+import { useMemo } from 'react';
 import { VscChevronDown, VscChevronRight, VscFile } from 'react-icons/vsc';
 
-import { FileEntry, FolderFileEntry } from '../atoms/types/FileEntry';
+import { FileExplorerEntry, FileExplorerFileEntry, FileExplorerFolderEntry } from '../atoms/types/FileExplorerEntry';
 import { FileNode } from './FileNode';
 
-interface FileEntryTreePropsBase {
-  files: FileEntry[];
+interface FileEntryTreeProps {
+  fileExplorerEntry: FileExplorerFolderEntry;
   onFileClick: (filePath: string) => void;
   selectedFiles: string[];
   paddingLeft?: string;
 }
 
-interface RootFileEntryTreeProps extends FileEntryTreePropsBase {
-  root: true;
-}
+export function FileEntryTree(props: FileEntryTreeProps) {
+  const { fileExplorerEntry, onFileClick, selectedFiles, paddingLeft = '0' } = props;
+  const files = useAtomValue(fileExplorerEntry.childrenAtom);
```

Comment:
nit: components usually are UI elements, without using atom. But I do understand how helpful it is for this scenario. Do you think we should move this file anywhere else or "relax" the rule?